### ISSUE TITLE
empty-lines rule parameters are ignored

### ISSUE THREAD

**Issue by g1ps** [adrienverge/yamllint#169]
**Environment**
Cygwin on Win 7 Pro
```
$ yamllint -v
yamllint 1.15.0
```

**YAML**
```

---

- hosts: some-hosts



  tasks:
  - name: some task
    import_role:
      name: some-role



````

**.yamllint**
```
extends: default

rules:
  empty-lines:
    max: 0
    max-end: 1
    max-start: 0
```

**Output**
Nil.

**Expected behaviour**
This should flag several places where multiple lines exist.

N.B The docs are inconsistent as to whether this format is correct/preferred or `empty-lines: {max: 0, max-end: 1, max-start: 0}. I tried both.

Thanks.

**Comment by adrienverge** [comment]
Hi,

I just tried, and it works fine for me. I have:
```
file.yaml
  2:1       error    too many blank lines (2 > 0)  (empty-lines)
  4:1       error    too many blank lines (1 > 0)  (empty-lines)
  8:1       error    too many blank lines (3 > 0)  (empty-lines)
  15:1      error    too many blank lines (3 > 1)  (empty-lines)
```

With:
- yamllint 1.15.0
- `file.yaml` and `.yamllint` in the current directory

**Comment by g1ps** [comment]
Tested again separately. Copied my yml above into a file. Using this config in `.yamllint`:
```

extends: default

rules:
  empty-lines:
    max: 0
    max-end: 1
    max-start: 0
  line-length:
    max: 250
  new-lines: disable  # Newlines might be either depending on the OS perforce checks out to.
  truthy: disable     # Ansible seems to have standardised on yes/no.
```
there is no output and nothing is flagged. What should I be looking at?

Thanks.

**Comment by adrienverge** [comment]
Could you test with #174? It aims to fix the Windows new-lines problem.

**Comment by adrienverge** [comment]
Closing because there was no feedback, and I cannot reproduce the problem.

**Comment by g1ps** [comment]
There's no feedback because I have no idea what you're asking. #174 didn't shed any light. You haven't clarified what the correct format is for the config entry. If you'd like to explain I can try whatever it is you're asking.

**Comment by anhuong** [comment]
I'm hitting a similar problem here. I have the config set to allow max 3 empty lines but it shows that error `3 > 0 (empty-lines)`

yamllint version 1.24.2

.yamllint
``yaml
extends: default

rules:
  braces:
    max-spaces-inside: 1
    level: error
  brackets:
    max-spaces-inside: 1
    level: error
  colons:
    max-spaces-after: -1
    level: error
  commas:
    max-spaces-after: -1
    level: error
  comments: disable
  comments-indentation: disable
  document-start: disable
  empty-lines:
    max: 10
    level: error
```

```sh
$ yamllint .
./defaults/main.yml
  49:1      error    too many blank lines (3 > 0)  (empty-lines)
```

Contents of `/defaults/main.yml`
```yaml
---
# defaults file for sch
default:
  sch:
    chart:
      app_name: ""
      label_type: non-prefixed
      secret_gen:
        suffix: "generated"
        secrets: []
    names:
      full_name:
        max_length: 63
        release_name_trunc_length: 42
        app_name_trunc_length: 20
        comp_name_trunc_length: 0
      full_comp_name:
        max_length: 63
        release_name_trunc_length: 36
        app_name_trunc_length: 13
        comp_name_trunc_length: 12
      stateful_set_name:
        max_length: 33
        release_name_trunc_length: 15
        app_name_trunc_length: 7
        comp_name_trunc_length: 9



```
Notice 3 lines at the end. Even when I change the config to allow max 10 empty lines or input the config via the `-d` flag in the commandline, I still see the same error. It looks like its still using the default value of 0.

**Comment by adrienverge** [comment]
@anhuong this isn't the same problem, and it looks like you didn't have a look at the documentation: https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.empty_lines
Is it max-end you're looking for?

**Comment by anhuong** [comment]
@adrienverge `max-end` does work for this case, but shouldn't `empty-lines{max: 3 }` also work? That means 3 consecutive lines are allowed in a file regardless of where they are. However, it is failing and claiming that the 3 empty lines is > 0.

**Comment by adrienverge** [comment]
@anhuong please read the documentation: https://yamllint.readthedocs.io/en/latest/rules.html#module-yamllint.rules.empty_lines

> `max-end` defines the maximal number of empty lines allowed at the end of the file. This option takes precedence over `max`.
>
> Default values (when enabled)
> ```
>      max-end: 0
> ```

**Comment by anhuong** [comment]
@adrienverge I have read the documentation and I am not specifying the `max-end`. If I don't specify it, does the default (0) still take precedence over the `max` value (3) that I am defining?

**Comment by harloprillar** [comment]
@anhuong yamllint uses Python dictionary update() function to extend default config with user-specified one: https://github.com/adrienverge/yamllint/blob/85ccd625a382423edd988fcb796c4dd5897886fb/yamllint/config.py#L66
, so if you not specified "rules.empty-lines.max-end" key in your config, it's not replaced default value, which is 0:
https://github.com/adrienverge/yamllint/blob/85ccd625a382423edd988fcb796c4dd5897886fb/yamllint/rules/empty_lines.py#L73
